components/raftstore/src/coprocessor/config.rs

```diff
@@ -103,12 +103,14 @@ impl Default for Config {
 
 impl Config {
     pub fn region_split_size(&self) -> ReadableSize {
```

Comment:
Re-reading this, I still think it's very confusing. Maybe something like this is better:
```rust
fn default_region_split_size(&self, raftstore_v2: bool) -> ReadableSize {
  if raftstore_v2 {
    RAFTSTORE_V2_SPLIT_SIZE
  } else if self.enable_region_bucket == Some(true) {
    LARGE_REGION_SPLIT_SIZE
  } else {
    SPLIT_SIZE
  }
}

pub fn region_split_size(&self) -> ReadableSize {
  self.region_split_size.unwrap_or(self.default_region_split_size(false))
}

pub fn optimize_for(&self, raftstore_v2: bool) {
  self.region_split_size.get_or_insert_with(|| self.default_region_split_size(raftstore_v2));
}
```

Reply:
Or perhaps we should not check for `enable_region_bucket` when computing default region size. `enable_region_bucket` should depend on the region size instead.

Reply:
I agree. In fact enable_region_bucket should only be used when bucket has a bug and needs to turn off explicitly. Otherwise it should be turned on by default when region size is large.